getBalances.js

```diff
@@ -1,77 +1,115 @@
-      logger.debug('calling homeBridge.methods.erc677token');
+    if (bridgeMode === BRIDGE_MODES.ERC_TO_ERC) {
+      const erc20Contract = new web3Foreign.eth.Contract(ERC20_ABI, ERC20_ADDRESS)
+      logger.debug('calling erc20Contract.methods.balanceOf')
+      const foreignErc20Balance = await erc20Contract.methods
+        .balanceOf(FOREIGN_BRIDGE_ADDRESS)
+        .call()
+      const homeBridge = new web3Home.eth.Contract(HOME_ERC_TO_ERC_ABI, HOME_BRIDGE_ADDRESS)
+      logger.debug('calling homeBridge.methods.erc677token')
       const tokenAddress = await homeBridge.methods.erc677token().call()
-      const tokenContract = new web3Home.eth.Contract(ERC677_ABI, tokenAddress);
-      logger.debug('calling tokenContract.methods.totalSupply()');
+      const tokenContract = new web3Home.eth.Contract(ERC677_ABI, tokenAddress)
+      logger.debug('calling tokenContract.methods.totalSupply()')
       const totalSupply = await tokenContract.methods.totalSupply().call()
       const foreignBalanceBN = new BN(foreignErc20Balance)
       const foreignTotalSupplyBN = new BN(totalSupply)
       const diff = foreignBalanceBN.minus(foreignTotalSupplyBN).toString(10)
-      logger.debug("Done");
+      logger.debug('Done')
       return {
         home: {
-          totalSupply: Web3Utils.fromWei(totalSupply),
+          totalSupply: Web3Utils.fromWei(totalSupply)
         },
         foreign: {
           erc20Balance: Web3Utils.fromWei(foreignErc20Balance)
         },
         balanceDiff: Number(Web3Utils.fromWei(diff)),
         lastChecked: Math.floor(Date.now() / 1000)
       }
-    } else {
-      logger.debug('calling web3Home.eth.getBalance');
+    } else if (bridgeMode === BRIDGE_MODES.NATIVE_TO_ERC) {
+      logger.debug('calling web3Home.eth.getBalance')
       const homeBalance = await web3Home.eth.getBalance(HOME_BRIDGE_ADDRESS)
-      const tokenContract = new web3Foreign.eth.Contract(ERC20_ABI, POA20_ADDRESS);
-      logger.debug('calling tokenContract.methods.totalSupply()');
+      const tokenContract = new web3Foreign.eth.Contract(ERC20_ABI, ERC20_ADDRESS)
+      logger.debug('calling tokenContract.methods.totalSupply()')
       const totalSupply = await tokenContract.methods.totalSupply().call()
       const homeBalanceBN = new BN(homeBalance)
       const foreignTotalSupplyBN = new BN(totalSupply)
       const diff = homeBalanceBN.minus(foreignTotalSupplyBN).toString(10)
-      logger.debug("Done");
+      logger.debug('Done')
       return {
         home: {
           balance: Web3Utils.fromWei(homeBalance)
         },
         foreign: {
-          totalSupply: Web3Utils.fromWei(totalSupply),
+          totalSupply: Web3Utils.fromWei(totalSupply)
         },
         balanceDiff: Number(Web3Utils.fromWei(diff)),
         lastChecked: Math.floor(Date.now() / 1000)
       }
+    } else if (bridgeMode === BRIDGE_MODES.ERC_TO_NATIVE) {
+      const erc20Contract = new web3Foreign.eth.Contract(ERC20_ABI, ERC20_ADDRESS)
```

Comment:
@akolotov @patitonar I think this is the most important part of the PR. Please check if the logic is OK.

Reply:
that's OK from my side.